pants

```diff
@@ -72,11 +89,17 @@ function get_pants_ini_config_value {
 # are installed and up to date.
 
 function determine_pants_runtime_python_version {
-  if which "${PYTHON_BIN_NAME}" >/dev/null; then
-    which "${PYTHON_BIN_NAME}"
+  if [[ "${PYTHON_BIN_NAME}" != 'unspecified' ]]; then
+    python_bin_name="${PYTHON_BIN_NAME}"
   else
-    die "Could not find ${PYTHON_BIN_NAME}. Please ensure ${PYTHON_BIN_NAME} is on your PATH."
+    interpreter_version="$(get_pants_ini_config_value 'pants_runtime_python_version')"
+    if [[ -z "${interpreter_version}" ]]; then
+      interpreter_version='2.7'
+    fi
+    python_bin_name="python${interpreter_version}"
   fi
+  check_interpreter_on_path "${python_bin_name}"
+  which "${python_bin_name}"
```

Comment:
If you remove the `>/dev/null` in `check_interpreter_on_path` and rename it to `get_exe_path_or_die` (which I would recommend anyway, as it doesn't do anything python-specific), then you can remove this `which`. Checking that an exe exists with `which`, but then returning its path in another `which` call feels error-prone to me (what if we slightly change this method later and we end up checking that one interpreter exists, but returning a path to another one?). The suggestion isn't about code golfing -- checking that something exists and then return a path to it separately feels error-prone. I also find the result (just calling `get_exe_path_or_die "${python_bin_name}"` at the end, which will output the exe path to stdout on success) extremely readable -- I don't know that the `check_interpreter_on_path` function could potentially exit from looking at its callsites.